flake.nix

```diff
@@ -1,42 +1,71 @@
 {
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/nixos-23.05";
-    nixpkgs-unstable.url = "github:NixOS/nixpkgs/nixos-unstable";
-    nixpkgs-kitman.url = "github:jkitman/nixpkgs/add-esplora-pkg";
+    nixpkgs = {
+      url = "github:nixos/nixpkgs/nixos-23.11";
+      # We use nixpkgs as input of `flakebox`, as it locks things like
+      # toolchains, in versions that are actually tested in flakebox's CI to
+      # cross-compile things well. This also saves us download and Nix
+      # evaluation time.
+    };
     flake-utils.url = "github:numtide/flake-utils";
     flakebox = {
-      url = "github:rustshop/flakebox?rev=390c23bc911b354f16db4d925dbe9b1f795308ed";
+      url = "github:rustshop/flakebox?rev=0d2e745bd8ca43ff0cb952debc4bea600e161508";
+      inputs.nixpkgs.follows = "nixpkgs";
     };
     advisory-db = {
       url = "github:rustsec/advisory-db";
       flake = false;
     };
   };
 
-  outputs = { self, nixpkgs, nixpkgs-unstable, nixpkgs-kitman, flake-utils, flakebox, advisory-db }:
+  outputs = { self, nixpkgs, flake-utils, flakebox, advisory-db }:
     flake-utils.lib.eachDefaultSystem
       (system:
         let
-          pkgs-unstable = import nixpkgs-unstable {
-            inherit system;
-          };
 
           pkgs = import nixpkgs {
             inherit system;
             overlays = [
               (final: prev: {
-                cargo-udeps = pkgs-unstable.cargo-udeps;
-                wasm-bindgen-cli = pkgs-unstable.wasm-bindgen-cli;
 
-                clightning = prev.clightning.overrideAttrs (oldAttrs: {
+                rocksdb_7_10 = prev.rocksdb_7_10.overrideAttrs (oldAttrs:
+                  pkgs.lib.optionalAttrs pkgs.stdenv.isDarwin {
+                    # C++ and its damn super-fragie compilation
+                    env = oldAttrs.env // {
+                      NIX_CFLAGS_COMPILE = oldAttrs.env.NIX_CFLAGS_COMPILE + " -Wno-error=unused-but-set-variable";
+                    };
+                  });
+
+                rocksdb_6_23 = prev.rocksdb_6_23.overrideAttrs (oldAttrs:
+                  pkgs.lib.optionalAttrs pkgs.stdenv.isDarwin {
+                    # C++ and its damn super-fragie compilation
+                    env = oldAttrs.env // {
+                      NIX_CFLAGS_COMPILE = oldAttrs.env.NIX_CFLAGS_COMPILE + " -Wno-error=unused-but-set-variable";
+                    };
+                  });
```

Comment:
~Why do we need two versions of rocksdb :open_mouth:~

I see, one for Fedimint and one for elecctrs.

Reply:
Nope, that would be too easy. One for electrs, one for esplora-electrs, and normal one for us. :D